src/scripts/main.js

```diff
@@ -1,3 +1,257 @@
+  } else if (!isSortingAscDesc) {
+    isSortingAscDesc = true;
+  } else if (isSortingAscDesc) {
+    isSortingAscDesc = false;
+  }
+
+  sortTableByColumnASC(index);
+});
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const Name = e.target.name.value;
+  const Position = e.target.position.value;
+  const Office = e.target.office.value;
+  const Age = e.target.age.value;
+  const Salary = e.target.salary.value;
+
+  if (Name.trim().length < 4) {
+    notification(
+      'error',
+      'Error',
+      'The Name field must have at least 5 characters.',
+    );
+
+    return;
+  }
+
+  if (+Age < 18 || +Age > 90) {
+    notification(
+      'error',
+      'Error',
+      'The Age field must be no less than 18 and no more than 90.',
+    );
+
+    return;
+  }
+
+  const salar = '$' + (+Salary).toFixed(3).replace(/\./g, ',');
+
+  const newEmployee = {
+    name: Name.trim(),
+    position: Position.trim(),
+    office: Office,
+    age: Age,
+    salary: salar,
+  };
+
+  const newTr = document.createElement('tr');
+
+  for (const key in newEmployee) {
+    const newTd = document.createElement('td');
+
+    newTd.textContent = newEmployee[key];
+    newTr.append(newTd);
+  }
+
+  tbody.append(newTr);
+
+  notification('succes', 'Success', 'New employee added successfully');
+
+  e.target.name.value = '';
+  e.target.position.value = '';
+  e.target.office.value = selectOffices[0];
+  e.target.age.value = '';
+  e.target.salary.value = '';
+});
+
+function sortTableByColumnASC(index) {
+  const activeRows = document.querySelectorAll('.active');
+
+  const sortArr = (index === 4)
+    ? sortBySalaryColumn(index)
+    : sortByRestColumns(index);
+
+  for (let y = 0; y < [...tbody.children].length; y++) {
+    for (let x = 0; x < [...tbody.children][y].children.length; x++) {
+      [...tbody.children][y].children[x].textContent = sortArr[y][x];
+    }
+
+    if (isActiveRow(y)) {
+      if ([...activeRows].length > 0) {
+        [...activeRows][0].classList.remove('active');
+      }
+
+      [...tbody.children][y].classList.add('active');
+    }
+  }
+}
+
+function sortBySalaryColumn(index) {
+  return (isSortingAscDesc)
+    ? [...tbody.children]
+      .map(r => [...r.children].map(c => c.textContent))
+      .sort((it1, it2) => (+it2[index].slice(1).replace(/,/g, '.'))
+          - (+it1[index].slice(1).replace(/,/g, '.')))
+    : [...tbody.children]
+      .map(r => [...r.children].map(c => c.textContent))
+      .sort((it1, it2) => (+it1[index].slice(1).replace(/,/g, '.'))
+          - (+it2[index].slice(1).replace(/,/g, '.')));
+}
+
+function sortByRestColumns(index) {
+  return (isSortingAscDesc)
+    ? [...tbody.children]
+      .map(r => [...r.children].map(c => c.textContent))
+      .sort((it1, it2) => it2[index].localeCompare(it1[index]))
+    : [...tbody.children]
+      .map(r => [...r.children].map(c => c.textContent))
+      .sort((it1, it2) => it1[index].localeCompare(it2[index]));
```

Comment:
here you're duplicating almost all the code. try to refactor it .
you can create a variable and assignt to it the result of map method. and then inside sort method use [destructuring assignment ](https://javascript.info/destructuring-assignment#array-destructuring) to swap values:


```javascript
const someValue = // the result of map method.
someValue.sort((a, b) => {
  const [it1, it2] = isSortingAscDesc ? [b, a] : [a, b]
  
  return it1[index].localeCompare(it2[index])
})
```